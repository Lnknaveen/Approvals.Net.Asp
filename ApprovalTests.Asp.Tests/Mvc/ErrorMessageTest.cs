using System.Web.Mvc;
using ApprovalTests.Asp.Mvc;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1.Controllers;

namespace ApprovalTests.Asp.Tests.Mvc
{
    [TestClass]
    [UseReporter(typeof (DiffReporter), typeof (FileLauncherReporter))]
    public class ErrorMessageTest : MvcTestBase
    {
        [TestMethod]
        public void TestMvcPage()
        {
            MvcApprovals.VerifyMvcPage<TestableNonExplicitController>(c => c.TestIndex);
        }
    }

    public class TestableNonExplicitController : TestableController<NonExplicitController>
    {
        public TestableNonExplicitController(NonExplicitController t)
            : base(t)
        {
        }

        public ActionResult TestIndex()
        {
            return ControllerUnderTest.Index();
        }
    }
}