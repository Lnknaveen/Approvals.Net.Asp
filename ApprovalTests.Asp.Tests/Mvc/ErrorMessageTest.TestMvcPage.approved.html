<pre>
Exception Thrown on Server.
If this is a 'View not Found' the most likely reason is your ActionResult is being calculated incorrectly. 

There are 2 ways to fix this.
1) Instead of extending Controller, extend ControllerWithExplicitViews
For Example:
   (wrong) ExampleController : Controller
   (right) ExampleController : ControllerWithExplicitViews
or 

2) add .Explicit() to your ViewResult

For Example:
 (wrong) return View();
 (right) return View().Explicit();

Message: The view 'TestIndex' or its master was not found or no view engine supports the searched locations. The following locations were searched:
~/Views/NonExplicit/TestIndex.aspx
~/Views/NonExplicit/TestIndex.ascx
~/Views/Shared/TestIndex.aspx
~/Views/Shared/TestIndex.ascx
~/Views/NonExplicit/TestIndex.cshtml
~/Views/NonExplicit/TestIndex.vbhtml
~/Views/Shared/TestIndex.cshtml
~/Views/Shared/TestIndex.vbhtml
</pre>